src/opentelemetry_instrumentation_kstreams/utils.py

```diff
@@ -83,10 +89,20 @@ def extract_send_topic(args, kwargs) -> Any:
         )
 
     @staticmethod
-    def extract_send_partition(args: Any, kwargs: Any) -> Any:
-        return KStreamsKafkaExtractor._extract_argument(
-            "partition", 4, None, args, kwargs
-        )
+    def extract_send_partition(record_metadata: Any) -> Any:
+        if hasattr(record_metadata, "partition"):
+            return record_metadata.partition
+
+    @staticmethod
+    def extract_send_offset(record_metadata: Any) -> Any:
+        if hasattr(record_metadata, "offset"):
```

Comment:
You can do `return `getattr(record_metadata, "offset", None)`, I think the return type shoud be `Optional[int]`